pkg/route/api/allocator/allocator.go

```diff
@@ -0,0 +1,52 @@
+package allocator
+
+import (
+	"fmt"
+	"regexp"
+
+	"code.google.com/p/go-uuid/uuid"
+	routeapi "github.com/openshift/origin/pkg/route/api"
+)
+
+// This should be something we get from config but we would still need a
+// default name if nothing's passed. The first pass uses the default name.
+// Would be better if we could use "v3.openshift.app", someone bought that!
+const DEFAULT_DNS_SUFFIX = "v3.openshift.com"
+
+// Invalid host name characters or not the <let-or-digit-or-hyphen> and
+// '.' separator as defined in rfc1123 (via rfc952)
+var name_RE, _ = regexp.Compile("[^A-Za-z0-9-.]+")
+
+func allocateShard(route *routeapi.Route) string {
+	//  For later, allocate names pointing to different shards.
+	shard := DEFAULT_DNS_SUFFIX
+
+	return shard
+}
+
+func generateName(route *routeapi.Route, suffix string) string {
+	name := route.ServiceName
+	if len(name) == 0 {
+		name = uuid.NewUUID().String()
+	}
+
+	var s = ""
+	if len(route.Namespace) > 0 {
+		s = fmt.Sprintf("%s-%s.%s", name, route.Namespace, suffix)
+	} else {
+		s = fmt.Sprintf("%s.%s", name, suffix)
+	}
+
+	return name_RE.ReplaceAllString(s, "")
```

Comment:
It is basically a regex that boils down to `const DNS1123LabelFmt string = "[a-z0-9]([-a-z0-9]*[a-z0-9])?"` with some other wrappers around it.  (k8s/pkg/util/validation.go)